### PR TITLE
all: Comments and cosmetics

### PR DESCRIPTION
**Pull request by acolomb**
Assortment of clean-ups and small fixes noted during recent PR activity.  See commit messages for explanation.

### Testing

Test suite passes locally.  Most of these should be no-ops.

### CODE REVIEW

**Comment by acolomb** [comment]
Feel free to point out unwanted parts.  Reverting single commits will be easy.

**Comment by acolomb** [comment]
Ping @syncthing/maintainers?

**Comment by calmh** [comment]
I'm OK with this apart from the added warning stuff. Those are IMHO just noise as they are not actionable; we should not add more warnings in the database layer.

File: lib/model/model.go [hunk 1]
```diff
@@ -3073,6 +3073,9 @@ func (m *model) DismissPendingDevice(device protocol.DeviceID) error {
 	l.Debugf("Discarding pending device %v", device)
 	err := m.db.RemovePendingDevice(device)
 	if err != nil {
+		msg := "Failed to remove pending device entry"
+		l.Warnf("%v: %v", msg, err)
+		m.evLogger.Log(events.Failure, msg)
 		return err
```
Comment (imsodin) [inline lib/model/model.go:3079]: Usually the pattern is that either a function handles the error (logging, events, actions, ...) or returns it - not both. Some in other instances. If you want/need to add context to the error, you can wrap it before returning.

Comment (acolomb) [reply lib/model/model.go:3079]: Well I was just making sure here that the handling is consistent whenever we call the function.  The code is the same as in `ccHandleFolders()`, so what should we do with that?

Regarding the warning being actionable, I think we're very unlikely to ever see it, but if so, it indicates some problem with the database, so would be valuable when trying to diagnose issues based on logs.

Comment (acolomb) [reply lib/model/model.go:3079]: Aw, scratch that.  I just overlooked that it is indeed reported through the API in these cases.

Comment (acolomb) [reply lib/model/model.go:3079]: Right now, the error in these cases will go totally unnoticed.  The JS side of things does not handle it, so the button just does nothing without any indication whatsoever.  Having at least a log entry is essential for diagnosis.

I tried adding `.error($scope.emitHTTPError)` to the respective `$http.delete()` call, but that just kills the GUI with the connection error panel?

Comment (calmh) [reply lib/model/model.go:3079]: Under what circumstances will an error occur, and what is the user expected to do about it?

Comment (acolomb) [reply lib/model/model.go:3079]: That's just the problem with just passing errors upwards. You never know about the details. No idea why a deletion in the DB should fail?

Of course, the user cannot do anything about it. But at least we will see in the provided logs that the backend has failed and it wasn't just a browser failure causing the "Dismiss" button to silently do nothing.

Comment (calmh) [reply lib/model/model.go:3079]: If the database has failed to the point where writes (deletes) return an error, we've already crashed with a myriad of cryptic messages. I wouldn't worry about it here.

Comment (acolomb) [reply lib/model/model.go]: Ok, removed the commit.

**Review by calmh — APPROVED** [review]

**Pull request merged**